packages/block-library/src/navigation-link/edit.js

```diff
@@ -242,6 +243,37 @@ export default function NavigationLinkEdit( {
 	const itemLabelPlaceholder = __( 'Add link…' );
 	const ref = useRef();
 
+	const isPostType =
+		( kind && 'post-type' === kind ) || ( type && 'post' === type );
+
+	const hasId = id && ! isNaN( id );
+	const postStatus = useSelect( ( select ) => {
+		if ( isPostType ) {
+			const { getEntityRecord } = select( coreStore );
+			const entityRecord = getEntityRecord( 'postType', type, id );
+
+			return entityRecord?.status;
+		}
+
+		return null;
+	} );
```

Comment:
We need to add the dependencies here: `[ isPostType, type, id ]`. Also let's revert the `if` here.